src/components/molecules/ImageInput/ImageInput.tsx

```diff
@@ -29,18 +35,49 @@ export const ImageInput = React.forwardRef<HTMLInputElement, ImageInputProps>(
       name,
       error,
       disabled,
-    } = props;
-
+      setValue,
+      register,
+    },
+    ref
+  ) => {
     const imageUrl = useMemo(
       () =>
         (image && image.length > 0 && URL.createObjectURL(image[0])) ||
         remoteImageUrl,
       [image, remoteImageUrl]
     );
 
+    const {
+      loading,
+      errorMessage,
+      handleFileInputChange,
+    } = useImageInputCompression(register, error?.message, name);
+
+    const handleFileInputChangeWrapper = useCallback(
+      async (event: ChangeEvent<HTMLInputElement>) => {
+        const [url, compressedFile] = await handleFileInputChange(event);
+        if (!compressedFile || !url) return;
+
+        setValue(name, [compressedFile], false);
+
+        if (remoteUrlInputName) {
+          setValue(remoteUrlInputName, url, true);
+        }
+      },
+      [handleFileInputChange, name, remoteUrlInputName, setValue]
+    );
+
     return (
       <>
-        <div className={`image-input default-container ${containerClassName}`}>
+        <div
+          className={classNames(
+            `image-input default-container`,
+            containerClassName,
+            {
+              disabled: loading,
```

Comment:
let's make it "image-input--disabled", m?

Reply:
There is also an opportunity to add `ImageInput` class at the block level and few `ImageInput__*` in the children level, besides the older classes.

They will not add or subtract from the style, but will be there for reference if a container of ImageInput needs to re-style some parts

Reply:
Yeah but... this `ImageInput` is all-legacy and I tried to match the code style there. 

Maybe it's better to address all the issues related to those `ImageInputs`s
- there two of them, merge them into one
- remove legacy styling
- try to simplify the `xxUrl` and `xxFile` props and forms

What do you think?